cmd/sign.go

```diff
@@ -0,0 +1,286 @@
+	  ]
+	}
+
+And the decoded JWT payload has the following form:
+
+	{
+	  "files": [
+		{
+		  "name": "foo/.manifest",
+		  "hash": "5716892ac6f2d6abc9417a65dbd01fcbd1b3d28afa2686d6378818052f78626f",
+		  "algorithm": "SHA256"
+		},
+		{
+		  "name": "foo/policy.rego",
+		  "hash": "a0d95905a421287c53299cfb601f0cabf205d35302e46da07b17f69bc7f2d7a6",
+		  "algorithm": "SHA256"
+		},
+		{
+		  "name": "foo/bar/data.json",
+		  "hash": "d78db014f90ba36982d08e0ce1c8c4d2511ab12196d104ec42ff2eb12c893b88",
+		  "algorithm": "SHA256"
+		}
+	  ]
+	}
+
+The "files" field is generated from the files under the directory path(s) provided by the --data flag.
+During bundle signature verification, OPA will check each file name (ex. "foo/bar/data.json") in the "files" field
+exists in the actual bundle. The file content is hashed using SHA256.
+
+To include additional claims in the payload use the --claims-file flag to provide a JSON file
+containing optional claims.
+
+For more information on the format of the ".signatures.json" file see
+https://www.openpolicyagent.org/docs/latest/management/#bundle-signature
+`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := doSign(cmdParams); err != nil {
+				fmt.Println("error:", err)
+				os.Exit(1)
+			}
+		},
+	}
+
+	signCommand.Flags().StringVarP(&cmdParams.key, "key", "k", "", "set the secret (HMAC) or path of the PEM file containing the private key (RSA and ECDSA)")
+	signCommand.Flags().StringVarP(&cmdParams.claimsFile, "claims-file", "", "", "set path of JSON file containing optional claims (see: https://openpolicyagent.org/docs/latest/management/#bundle-signature-format)")
+	signCommand.Flags().StringVarP(&cmdParams.outputFilePath, "output-file-path", "o", ".", "set the location for the .signatures.json file")
+	addSigningAlgFlag(signCommand.Flags(), &cmdParams.algorithm, defaultTokenSigningAlg)
+	addDataFlag(signCommand.Flags(), &cmdParams.dataPaths)
+
+	RootCommand.AddCommand(signCommand)
+}
+
+func doSign(params signCmdParams) error {
+	if len(params.dataPaths.v) == 0 {
+		return fmt.Errorf("specify atleast one directory containing policy and data files")
+	}
+
+	if params.key == "" {
+		return fmt.Errorf("specify the secret (HMAC) or path of the PEM file containing the private key (RSA and ECDSA)")
+	}
+
+	payload, err := generatePayload(params.dataPaths.v, params.claimsFile)
+	if err != nil {
+		return err
+	}
+
+	privKey, err := getPrivateKey(params.key, params.algorithm)
+	if err != nil {
+		return err
+	}
+
+	// generate signed token
+	token, err := jws.SignWithOption(payload, jwa.SignatureAlgorithm(params.algorithm), privKey)
+	if err != nil {
+		return err
+	}
+
+	return writeTokenToFile(string(token), params.outputFilePath)
+}
+
+func generatePayload(dataPaths []string, claimsFile string) ([]byte, error) {
```

Comment:
Let's figure out how to avoid this. We don't want to duplicate this code. We'll likely need to improve/refactor the loader package.